test_runner/random_test_runner/launch/random_test.launch.py

```diff
@@ -32,26 +32,45 @@
 from launch.substitutions import LaunchConfiguration
 from launch_ros.actions import Node
 
+
+def architecture_types():
+    return ["awf/auto", "awf/universe", "tier4/proposal"]
+
+
 def default_autoware_launch_package_of(architecture_type):
+    if architecture_type not in architecture_types():
+        raise KeyError(
+            f"architecture_type:={architecture_type.perform(context)} is not supported. Choose one of {architecture_types()}."
+        )
+
```

Comment:
If I remember correctly this check is redundant since possible values are already checked by the launch system. It is verified through passing possible values in DeclareLaunchArgument
It does not do any harm, but probably will never be reached.

Reply:
But firstly, `KeyError` is raised in `default_autoware_launch_package_of` function which is not very verbose:
```
KeyError: 'test/architecture'
```